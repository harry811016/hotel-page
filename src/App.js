import React, { Fragment, useState, useEffect } from 'react'
import Navbar from './components/Navbar/Navbar'
import Card from './components/Card/Card'
import hotelApis from './hooks/useHotelAPI'

const useHotelAPI = hotelApis.useHotelAPI

const App = () => {
  const [currency, setCurrency] = useState('USD')
  const [hoteldata, fetchHoteldata] = useHotelAPI(currency)

  const handleCurrencyChange = (e) => {
    setCurrency(e.target.value)
    window.localStorage.setItem('hotelPageCurrency', e.target.value)
  }

  useEffect(() => {
    setCurrency(window.localStorage.getItem('hotelPageCurrency') ? window.localStorage.getItem('hotelPageCurrency') : 'USD')
    fetchHoteldata(currency)
  }, [currency, fetchHoteldata])

  return (
    <Fragment>
      <Navbar handleCurrencyChange={handleCurrencyChange} currency={currency} />
      <Card currency={currency} hoteldata={hoteldata} />
    </Fragment>
  )
}

export default App
