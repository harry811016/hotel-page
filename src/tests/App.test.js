import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { ThemeProvider } from 'emotion-theming'
import App from '../App'

afterEach(cleanup)

it('should take a snapshot', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />)
  const result = renderer.getRenderOutput()
  expect(result).toMatchSnapshot()
})