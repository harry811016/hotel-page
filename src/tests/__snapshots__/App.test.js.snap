// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<React.Fragment>
  <Navbar
    currency="USD"
    handleCurrencyChange={[Function]}
  />
  <Card
    hoteldata={
      Object {
        "currency": "USD",
        "data": Array [],
        "isLoading": true,
      }
    }
  />
</React.Fragment>
`;
